Content/Items/Misc/Weapons.GeodeBow.cs

```diff
@@ -0,0 +1,482 @@
+            color.A = 0;
+            Main.spriteBatch.Draw(glowTex, Projectile.Center - Main.screenPosition, null, color, Projectile.rotation, glowTex.Size() / 2, 1 + (0.1f * (float)Math.Sin(pulseCounter)), SpriteEffects.None, 0f);
+            return false;
+        }
+
+        private void Shatter(Projectile proj)
+        {
+            proj.penetrate--;
+            Helper.PlayPitched("Impacts/GlassExplodeShort", 1, Main.rand.NextFloat(0.1f, 0.3f), Projectile.Center);
+
+            int counter = 0;
+
+            foreach (Projectile otherCrystal in Main.projectile)
+            {
+                if (otherCrystal.active && otherCrystal.type == Projectile.type && (otherCrystal.ModProjectile as GeodeBowGrowth).target == target)
+                    counter++;
+
+                if (counter > 6)
+                    break;
+            }
+
+            Core.Systems.CameraSystem.Shake += (int)Math.Min(11, 5 * (float)Math.Sqrt(counter));
+
+            for (int k = 0; k < 12; k++)
+                Dust.NewDustPerfect(Projectile.Center + Main.rand.NextVector2Circular(30,30), ModContent.DustType<Dusts.ArtifactSparkles.GeodeArtifactSparkleFast>(), Main.rand.NextVector2Circular(7, 7), 0, default, Main.rand.NextFloat(0.85f, 1.15f));
+
+            DistortionPointHandler.AddPoint(Projectile.Center, (float)Math.Sqrt(counter), 0.5f,
+                    (intensity, ticksPassed) => intensity,
+                    (progress, ticksPassed) => (float)Math.Sqrt(ticksPassed / 14f) + 0.5f,
+                    (progress, intensity, ticksPassed) => ticksPassed <= 7);
+
+            Projectile.NewProjectile(Projectile.GetSource_FromThis(), Projectile.Center, Vector2.Zero, ModContent.ProjectileType<GeodeBowExplosion>(), (int)(Math.Pow(counter, 0.7f) * Projectile.damage), Projectile.knockBack, owner.whoAmI, 0, (int)Math.Sqrt(counter) * 0.5f);
+
+            for (int j = 0; j < 16; j++)
+            {
+                Vector2 direction = Main.rand.NextVector2CircularEdge(1, 1);
+                Projectile shrapnel = Projectile.NewProjectileDirect(Projectile.GetSource_FromThis(), Projectile.Center + (direction * 10), direction * Main.rand.Next(5, 10) * (float)Math.Sqrt(counter), ModContent.ProjectileType<GeodeBowShrapnel>(), 0, 0, owner.whoAmI);
+                (shrapnel.ModProjectile as GeodeBowShrapnel).width = Main.rand.Next(30, 50);
+                shrapnel.timeLeft = Main.rand.Next(90);
+            }
+
+            Vector2 vel = Vector2.Normalize(offset) * 5;
+            for (int i = 1; i <= 4; i++)
+                Gore.NewGore(Projectile.GetSource_FromThis(), Projectile.Center, vel.RotatedByRandom(0.4f), Mod.Find<ModGore>("GeodeBowGrowthGore" + i.ToString()).Type, Projectile.scale);
+            Projectile.active = false;
+        }
+    }
+
+    internal class GeodeBowExplosion : ModProjectile
+    {
+        public override string Texture => AssetDirectory.Invisible;
+
+        public float radiusMult => Projectile.ai[1];
```

Comment:
use a ``ref`` property for aliasing AI fields so you can both get and set with the friendly name

Reply:
Fixed